src/api/Tron.js

```diff
@@ -185,3 +200,197 @@ export const validateAddress = async (address: string) => {
   }
   return false;
 };
+
+// cache for account names (name is unchanged over time)
+const accountNamesCache = makeLRUCache(
+  async (addr: string): Promise<?string> => getAccountName(addr),
+  (addr: string) => addr,
+  {
+    max: 300,
+    maxAge: 180 * 60 * 1000 // 3hours
+  }
+);
+
+// cache for super representative brokerages (brokerage is unchanged over time)
+const srBrokeragesCache = makeLRUCache(
+  async (addr: string): Promise<number> => getBrokerage(addr),
+  (addr: string) => addr,
+  {
+    max: 300,
+    maxAge: 180 * 60 * 1000 // 3hours
+  }
+);
+
+export const getAccountName = async (addr: string): Promise<?string> => {
+  const tronAcc = await fetchTronAccount(addr);
+  const acc = tronAcc[0];
+  const accountName: ?string = acc.account_name 
+    ? hexToAscii(acc.account_name) 
+    : undefined;
+
+  accountNamesCache.hydrate(addr, accountName); // put it in cache
+
+  return accountName;
+};
+
+export const getBrokerage = async (addr: string): Promise<number> => {
+  const { brokerage } = await post(`${baseApiUrl}/wallet/getBrokerage`, { address: addr });
+
+  srBrokeragesCache.hydrate(addr, brokerage); // put it in cache
+
+  return brokerage;
+};
+
+export const getTronSuperRepresentatives = async (): Promise<SuperRepresentative[]> => {
+  try {
+    const result = await post(`${baseApiUrl}/wallet/listwitnesses`, {});
+    
+    const superRepresentatives = 
+      await Promise.all(
+        result.witnesses.map(w => {
+          const encodedAddress = encode58Check(w.address);
+
+          return accountNamesCache(encodedAddress).then(accountName =>
+            srBrokeragesCache(encodedAddress).then(brokerage => (
+              {
+                ...w,
+                address: encodedAddress,
+                name: accountName,
+                brokerage,
+                voteCount: w.voteCount || 0,
+                isJobs: w.isJobs || false
+              }
+            ))
+          );
+        })
+      );
+
+    const sortedSrs = superRepresentatives.sort((a, b) => b.voteCount - a.voteCount);
+
+    return sortedSrs;
+  } catch (e) {
+    throw new Error("Unexpected error occured when calling getTronSuperRepresentatives");
+  }
+};
+
+export const getNextVotingDate = async (): Promise<Date> => {
+  const { num } = await post(`${baseApiUrl}/wallet/getnextmaintenancetime`);
+  return new Date(num);
+};
+
+export const getTronSuperRepresentativeData = async (max: ?number): Promise<SuperRepresentativeData> => {
```

Comment:
Yeah, because position (sorted by vote count) and metadata like block total produced constantly changed.
By the way, fetching list from trongrid is rather fast, you could try it there: https://developers.tron.network/reference#walletlistwitnesses

Otherwise, I put in cache super representative names and their brokerage value (2 dedicated calls) and theses data never change.